### PR TITLE
Update checkout pricing when changing shipping methods

### LINKED ISSUE #10828: Changing shipping method doesn't change price
**Sylius version affected**: All I suppose

**Description**  
image

When I choose another shipping method, the shipping costs are not updated and neither the total obviously.

**Possible Solution**  
A javascript solution should suffice

### PR DESCRIPTION
**Pull request by TheMelter**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #10828 
| License         | MIT

ezgif com-video-to-gif (1)
ezgif com-video-to-gif

### CODE REVIEW

**Comment by vvasiloi** [comment]
This has to be done through AJAX with proper formatting of the amount in backend.
Also, changing the shipping method might affect the totals, like apply/revert a promotion.

File: features/checkout/shipping_order/selecting_order_shipping_method.feature [hunk 1]
```diff
@@ -17,3 +17,14 @@ Feature: Selecting order shipping method
...
         And I complete the shipping step
+        Then I should see shipping total "$10.00"
+        And I should see total price "$29.99"
+
+    @ui @javascript
+    Scenario: Changing one of available shipping method
+        Given I have product "Targaryen T-Shirt" in the cart
+        And I specified the shipping address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        When I select "Raven Post" shipping method
+        And I select "Dragon Post" shipping method
```
[the remark this answers is not in the document]
Comment (TheMelter) [reply features/checkout/shipping_order/selecting_order_shipping_method.feature]: Updated!

**Comment by TheMelter** [comment]
@vvasiloi You're right. I've adjusted the code to use AJAX to update the checkout summary instead. Please take a look.

File: features/checkout/shipping_order/selecting_order_shipping_method.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheMelter) [reply features/checkout/shipping_order/selecting_order_shipping_method.feature]: Added screenshots as well.

File: src/Sylius/Behat/Page/Shop/Checkout/SelectShippingPageInterface.php [hunk 2]
```diff
@@ -81,4 +81,14 @@ public function isNextStepButtonUnavailable();
...
+
+    /**
+     * @return string
+     */
+    public function getShippingTotal();
+
+    /**
+     * @return string
+     */
+    public function getTotalPrice();
```
Comment (vvasiloi) [inline src/Sylius/Behat/Page/Shop/Checkout/SelectShippingPageInterface.php]: It's tehnically a BC break, @pamil?

File: src/Sylius/Bundle/ShopBundle/Resources/config/routing/ajax.yml [hunk 3]
```diff
@@ -9,6 +9,12 @@ sylius_shop_ajax_cart:
     resource: "@SyliusShopBundle/Resources/config/routing/ajax/cart.yml"
     prefix: /cart
 
+sylius_shop_ajax_shipping_fee:
+    path: /shipping-fee/{shippingMethodId}
```
Comment (vvasiloi) [inline src/Sylius/Bundle/ShopBundle/Resources/config/routing/ajax.yml]: Better use the code. Usage of IDs in front-end is generally avoided.

Comment (TheMelter) [reply src/Sylius/Bundle/ShopBundle/Resources/config/routing/ajax.yml]: Fixed

File: src/Sylius/Behat/Page/Shop/Checkout/SelectShippingPageInterface.php (hunk 2 above)
Comment (Zales0123) [reply src/Sylius/Behat/Page/Shop/Checkout/SelectShippingPageInterface.php]: There is not BC policy on Behat classes/services, so it should not be a problem ;)

**Pull request closed without merging**